sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/DataSourceResolution.scala

```diff
@@ -296,6 +310,17 @@ case class DataSourceResolution(
       orCreate = replace.orCreate)
   }
 
+  private def convertDeleteFrom(
+      catalog: TableCatalog,
+      identifier: Identifier,
+      delete: DeleteFromStatement): DeleteFromTable = {
+    val relation = CatalogV2Util.loadTable(catalog, identifier)
+        .map(DataSourceV2Relation.create).getOrElse(UnresolvedRelation(delete.tableName))
```

Comment:
shall we fail here if the table not found?

Reply:
I don't think this should throw an exception. We want `CheckAnalysis` to do that. I also don't think this should resolve the relation because `ResolveTables` is responsible for that, to keep the lookup code in one place instead of adding it in lots of places.

This should not call `loadTable` and should instead insert the `UnresolvedRelation` from the `getOrElse` case. Then `ResolveTables` will load the table and `CheckAnalysis` will throw an exception if the table isn't found.

Reply:
Thank you @cloud-fan @rdblue . Have updated the code and pls take a look.